daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -642,94 +643,171 @@ private[lf] object Speedy {
-            SGenMap(
-              isTextMap = false,
-              entries = entries.iterator.map { case (k, v) => go(k) -> go(v) },
-            )
-          case V.ValueVariant(Some(id), variant, arg) =>
-            compiledPackages.getSignature(id.packageId) match {
-              case Some(pkg) =>
-                pkg.lookupDefinition(id.qualifiedName).fold(crash, identity) match {
-                  case DDataType(_, _, data: DataVariant) =>
-                    SVariant(id, variant, data.constructorRank(variant), go(arg))
+    private[speedy] def importValue(typ0: Type, value0: V[V.ContractId]): Unit = {
+
+      def go(ty: Type, value: V[V.ContractId]): SValue = {
+        def typeMismatch = crash(s"mismatching type: $ty and value: $value")
+
+        val (tyFun, tyArgs) = destructApp(ty)
+        tyFun match {
+          case TBuiltin(_) =>
+            tyArgs match {
+              case Nil =>
+                value match {
+                  case V.ValueInt64(value) =>
+                    SValue.SInt64(value)
+                  case V.ValueNumeric(value) =>
+                    SValue.SNumeric(value)
+                  case V.ValueText(value) =>
+                    SValue.SText(value)
+                  case V.ValueTimestamp(value) =>
+                    SValue.STimestamp(value)
+                  case V.ValueDate(value) =>
+                    SValue.SDate(value)
+                  case V.ValueParty(value) =>
+                    SValue.SParty(value)
+                  case V.ValueBool(b) =>
+                    if (b) SValue.SValue.True else SValue.SValue.False
+                  case V.ValueUnit =>
+                    SValue.SValue.Unit
                   case _ =>
-                    crash(s"definition for variant $id not found")
+                    typeMismatch
                 }
-              case None =>
-                throw SpeedyHungry(
-                  SResultNeedPackage(
-                    id.packageId,
-                    pkg => {
-                      compiledPackages = pkg
-                      returnValue = go(value)
-                    },
-                  )
-                )
-            }
-          case V.ValueEnum(Some(id), constructor) =>
-            compiledPackages.getSignature(id.packageId) match {
-              case Some(pkg) =>
-                pkg.lookupDefinition(id.qualifiedName).fold(crash, identity) match {
-                  case DDataType(_, _, data: DataEnum) =>
-                    SEnum(id, constructor, data.constructorRank(constructor))
+              case elemType :: Nil =>
+                value match {
+                  case V.ValueContractId(cid) =>
+                    addGlobalCid(cid)
+                    SValue.SContractId(cid)
+                  case V.ValueNumeric(d) =>
+                    elemType match {
+                      case TNat(s) =>
+                        data.Numeric.fromBigDecimal(s, d).fold(crash, SValue.SNumeric)
```

Comment:
I think I’m slightly confused about this. Why do we need to handle this differently now than we handled it before? We don’t need to augment numerics with type info do we?